src/go/MONIT/alert.go

```diff
@@ -0,0 +1,571 @@
+		if filteredAlerts[each.Name] == 1 {
+			fmt.Fprintf(w, " %s\t\t%s\t\t%s\t\t%s\t\t%s",
+				each.Name,
+				each.Service,
+				each.Tag,
+				each.Severity,
+				timeDiff(time.Now(), each.StartsAt, 0),
+			)
+			if each.EndsAt == time.Unix(maxtstmp, 0).UTC() {
+				fmt.Fprintf(w, "\t\t%s", "Undefined")
+				fmt.Fprintf(w, "\t\t%s\n", "Undefined")
+			} else {
+				fmt.Fprintf(w, "\t\t%s", timeDiff(time.Now(), each.EndsAt, 0))
+				fmt.Fprintf(w, "\t\t%s\n", timeDiff(each.StartsAt, each.EndsAt, 1))
+			}
+
+		}
+	}
+}
+
+//Function for printing alert's details in JSON format
+func jsonPrintDetails() {
+
+	b, err := json.Marshal(alertDetails[name])
+
+	if err != nil {
+		log.Printf("Unable to convert Detailed JSON Data, error: %v\n", err)
+		return
+	}
+
+	fmt.Println(string(b))
+}
+
+//Function to get all keys of type map[string]interface{}
+func getkeys(m map[string]interface{}) []string {
+
+	keys := make([]string, 0, len(m))
+	for key := range m {
+		keys = append(keys, key)
+	}
+	return keys
+}
+
+//Function for printing alert's details in Plain text format
+func detailPrint() {
+
+	if alert, ok := alertDetails[name]; ok {
+		labelsKeys := getkeys(alertDetails[name].Labels)
+		sort.Strings(labelsKeys)
+
+		for _, each := range labelsKeys {
+			switch each {
+			case "alertname":
+				fmt.Printf("%s: %s\n", configJSON.Names[0], alertDetails[name].Labels[each])
+				fmt.Printf("%s\n", configJSON.Names[1])
+			case "service", "tag", "severity":
+				fmt.Printf("\t%s: %s\n", each, alertDetails[name].Labels[each])
+			}
+		}
+
+		fmt.Printf("%s\n", configJSON.Names[2])
+		for key, value := range alert.Annotations {
+			fmt.Printf("\t%s: %s\n", key, value)
+		}
+	} else {
+		fmt.Printf("%s alert not found\n", name)
+	}
+
+}
+
+//Function running all logics
+func run() {
+
+	var amdata amData
+	get(&amdata)
+	mergeData(amdata)
+	sortAlert()
+	filter()
+
+	if *jsonOutput {
+		if *details {
+			jsonPrintDetails()
+		} else {
+			jsonPrint()
+		}
+	} else {
+		if *details {
+			detailPrint()
+		} else {
+			tabulate()
+		}
+	}
+}
+
+func parseConfig() {
+
+	configFile = os.Getenv("CONFIG_PATH") //CONFIG_PATH Environment Variable storing config filepath.
+
+	//Defaults in case no config file is provided
+	configJSON.CMSMONURL = "https://cms-monitoring.cern.ch"
+	configJSON.Names = []string{"NAMES", "LABELS", "ANNOTATIONS"}
+	configJSON.Columns = []string{"NAME", "SERVICE", "TAG", "SEVERITY", "STARTS", "ENDS", "DURATION"}
+	configJSON.Verbose = 0
+	configJSON.SeverityLevels = make(map[string]int)
+	configJSON.SeverityLevels["info"] = 0
+	configJSON.SeverityLevels["warning"] = 1
+	configJSON.SeverityLevels["medium"] = 2
+
+	if configFile != "" {
+		jsonFile, e := os.Open(configFile)
+		if e != nil {
+			fmt.Println("Config File not found, error:", e)
+		}
+		defer jsonFile.Close()
+		decoder := json.NewDecoder(jsonFile)
+		err := decoder.Decode(&configJSON)
+		if err != nil {
+			fmt.Println("Config JSON File can't be loaded, error:", err)
```

Comment:
I think it would be better to stop execution of cli if config file is broken, just use fmt.Fatalf

Reply:
Sure.